### PR TITLE
Configure CKEditor 4 with XBlock

### PR DESCRIPTION
**Pull request by rhn2**

### CODE REVIEW

File: journal_xblock/journal_xblock.py [hunk 1]
```diff
@@ -45,10 +49,13 @@ def student_view(self, context=None):
         The primary view of the JournalXBlock, shown to students
         when viewing courses.
         """
-        html = render_template("static/html/journal_xblock.html", {"self": self})
-        frag = Fragment(html.format(self=self))
+        html = render_template("static/html/journal_xblock.html", {"self": self,
+                                                                   "is_studio": getattr(self.system, "is_author_mode", False)})
```
Comment (m-usman01) [inline journal_xblock/journal_xblock.py:53]: Why is "is_studio" needed in LMS?

File: journal_xblock/journal_xblock.py [hunk 2]
```diff
@@ -58,9 +65,10 @@ def studio_view(self, context=None):
         when editing xblock.
         """
         html = render_template("static/html/journal_xblock_studio.html", {"self": self})
-        frag = Fragment(html.format(self=self))
+        frag = Fragment(html)
         frag.add_css(self.resource_string("static/css/journal_xblock.css"))
-        frag.add_javascript(self.resource_string("static/js/src/journal_xblock.js"))
+        frag.add_javascript(_resource("static/js/src/journal_xblock.js"))
```
Comment (m-usman01) [inline journal_xblock/journal_xblock.py:70]: Can we use a separate JS file for studio and LMS?

File: journal_xblock/journal_xblock.py [hunk 3]
```diff
@@ -90,3 +98,11 @@ def workbench_scenarios():
...
              """),
         ]
+
+
+def _resource(path):  # pragma: NO COVER
+    """
+    Handy helper for getting resources from our kit.
+    """
+    data = pkg_resources.resource_string(__name__, path)
+    return data.decode("utf8")
```
Comment (m-usman01) [inline journal_xblock/journal_xblock.py:108]: Missing EOF

File: journal_xblock/static/css/journal_xblock.css [hunk 4]
```diff
@@ -3,3 +3,55 @@
...
+
+#journal_log .card .collapse {
+  border: 0;
+}
+
+#journal_log .card .collapse.show {
+  line-height: 30px;
+  color: #222;
+    margin-top: 2%;
+}
```
Comment (m-usman01) [inline journal_xblock/static/css/journal_xblock.css:57]: Missing EOF

File: journal_xblock/static/html/journal_xblock_studio.html [hunk 5]
```diff
@@ -62,4 +61,8 @@
       </li>
     </ul>
   </div>
+    <script type="text/javascript">
+        CKEDITOR.replace( 'studio_editor').setData($("#studio_editor").attr("data-value"));
+    </script>
+
 </div>
```
Comment (m-usman01) [inline journal_xblock/static/html/journal_xblock_studio.html:68]: Missing EOF

File: journal_xblock/journal_xblock.py (hunk 2 above)
Comment (rhn2) [reply journal_xblock/journal_xblock.py:70]: There isn't much JS required in studio. Only JS part which required is Editor setting and that is general for both lms and studio. So, it'll be extra file with no use.

File: journal_xblock/journal_xblock.py (hunk 1 above)
Comment (rhn2) [reply journal_xblock/journal_xblock.py:53]: It's for the studio side when learners add Journal Block. Make the editor un-editable for lms view on studio side

**Review by m-usman01 — APPROVED** [review]

**Pull request merged**